be/src/exprs/expr.cpp

```diff
@@ -453,117 +428,7 @@ Status Expr::create_expr(ObjectPool* pool, const TExprNode& texpr_node, Expr** e
-        *expr = pool->add(new NullLiteral(texpr_node));
-        break;
-    case TExprNodeType::ARITHMETIC_EXPR:
-        if (texpr_node.opcode != TExprOpcode::INVALID_OPCODE) {
-            *expr = pool->add(ArithmeticExpr::from_thrift(texpr_node));
-            break;
-        }
-    case TExprNodeType::CAST_EXPR:
-        if (texpr_node.__isset.child_type) {
-            *expr = pool->add(CastExpr::from_thrift(texpr_node));
-            break;
-        }
-    case TExprNodeType::COMPUTE_FUNCTION_CALL:
-    case TExprNodeType::FUNCTION_CALL:
-        DCHECK(texpr_node.__isset.fn);
-
-        if (texpr_node.fn.name.function_name == "if") {
-            *expr = pool->add(new IfExpr(texpr_node));
-        } else if (texpr_node.fn.name.function_name == "nullif") {
-            *expr = pool->add(new NullIfExpr(texpr_node));
-        } else if (texpr_node.fn.name.function_name == "ifnull") {
-            *expr = pool->add(new IfNullExpr(texpr_node));
-        } else if (texpr_node.fn.name.function_name == "coalesce") {
-            *expr = pool->add(new CoalesceExpr(texpr_node));
-        } else {
-            *expr = pool->add(new ScalarFnCall(texpr_node));
-        }
-        break;
-    case TExprNodeType::CASE_EXPR: {
-        if (!texpr_node.__isset.case_expr) {
-            return Status::InternalError("Case expression not set in thrift node");
-        }
-
-        *expr = pool->add(new CaseExpr(texpr_node));
-        break;
-    }
-    case TExprNodeType::IN_PRED: {
-        switch (texpr_node.opcode) {
-        case TExprOpcode::FILTER_IN:
-        case TExprOpcode::FILTER_NOT_IN:
-            *expr = pool->add(new InPredicate(texpr_node));
-            break;
-        default:
-            *expr = pool->add(new ScalarFnCall(texpr_node));
-            break;
-        }
-        break;
-    }
-    case TExprNodeType::SLOT_REF: {
-        if (!texpr_node.__isset.slot_ref) {
-            return Status::InternalError("Slot reference not set in thrift node");
-        }
-
-        *expr = pool->add(new SlotRef(texpr_node));
-        break;
-    }
-    case TExprNodeType::TUPLE_IS_NULL_PRED: {
-        *expr = pool->add(new TupleIsNullPredicate(texpr_node));
-        break;
-    }
-    case TExprNodeType::INFO_FUNC: {
-        *expr = pool->add(new InfoFunc(texpr_node));
-        break;
-    }
-    case TExprNodeType::AGG_EXPR:
-    case TExprNodeType::TABLE_FUNCTION_EXPR:
-    case TExprNodeType::IS_NULL_PRED:
-    case TExprNodeType::LIKE_PRED:
-    case TExprNodeType::LITERAL_PRED:
-    case TExprNodeType::ARRAY_EXPR:
-    case TExprNodeType::ARRAY_ELEMENT_EXPR:
-    case TExprNodeType::ARRAY_SLICE_EXPR:
-        break;
-    }
-
-    if (*expr == nullptr) {
-        LOG(WARNING) << "Engine node type return nullptr: " + std::to_string(texpr_node.node_type);
-        return Status::InternalError("Engine not support the operator");
-    }
-    return Status::OK();
+    return Status::InternalError("Don't support old query engine");
```

Comment:
I think it's better use "Does not support old query engine any more."

Reply:
OK